doc/storage/api/api.rst

```diff
@@ -313,6 +356,135 @@ These definitions must be defined in the header file :file:`psa/internal_trusted
 
     * Even if all parameters are correct, the function can fail in the case of a storage failure.
 
+.. function:: psa_its_iterator_start
+
+   .. summary::
+       Initializes an iterator that can be used to return a list of ``uid`` values in the Internal Trusted Storage.
+       
+       This function must be fully defined if `PSA_STORAGE_SUPPORT_ITERATION` is true.
+       
+       If `PSA_STORAGE_SUPPORT_ITERATION` is false, then this function SHALL always return ``PSA_ERROR_NOT_SUPPORTED``
+
+
+   .. param::  psa_its_storage_iterator_t* context
+       A pointer to a context for this iterator. The pointer may be NULL. This is set to a new value on success and is undefined on error. The content of the iterator is implementation defined.
```

Comment:
The optional `NULL` value looks odd, as it would seem that the entire purpose of starting an iteration would be to iterate.

There is no mention of how the function behaves if no iterator object is provided, or why a caller would ever want to do this. If there is a use case (I can think of a couple) where this is a better approach than provided an iterator object, it would benefit from an explanation and/or code example in the description.

Reply:
The same applies to `psa_ps_iterator_start()`.